performance/class-mime-types-caching.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * MIME Types Caching.
+ *
+ * @package vip-performance
+ */
+
+// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery
+
+namespace Automattic\VIP\Performance;
+
+/**
+ * MIME Types Caching class.
+ */
+class Mime_Types_Caching {
+	public const MINIMUM_WORDPRESS_VERSION  = '6.4';
+	public const CACHE_GROUP                = 'mime_types';
+	public const MIME_TYPES_CACHE_KEY       = 'vip_mime_types';
+	public const MAX_POSTS_TO_QUERY_DEFAULT = 500000;
+
+	/**
+	 * Class initialization.
+	 */
+	public static function init() {
+		global $wp_version;
+
+		if ( isset( $wp_version ) &&
+			version_compare( $wp_version, self::MINIMUM_WORDPRESS_VERSION, '>=' ) ) {
+			static::enable_post_mime_types_caching();
```

Comment:
The mechanism to turn this off would be removing the action from `init`:
```
remove_action( 'init', 'vip_cache_mime_types' );
```
This approach was Caleb's suggestion from a [previous review](https://github.com/Automattic/vip-go-mu-plugins/pull/4953#discussion_r1382281936) instead of using a new filter.